integration-tests/e2e_tests_ci_tool/cmd/filter_cmd.go

```diff
@@ -0,0 +1,150 @@
+package cmd
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+)
+
+// Filter tests based on workflow, test type, and test IDs.
+func filterTests(allTests []CITestConf, workflow, testType, ids string) []CITestConf {
+	workflowFilter := workflow
+	typeFilter := testType
+	idFilter := strings.Split(ids, ",")
+
+	var filteredTests []CITestConf
+
+	for _, test := range allTests {
+		workflowMatch := workflow == "" || contains(test.Workflows, workflowFilter)
+		typeMatch := testType == "" || test.TestType == typeFilter
+		idMatch := ids == "*" || ids == "" || contains(idFilter, test.ID)
+
+		if workflowMatch && typeMatch && idMatch {
+			test.IDSanitized = sanitizeTestID(test.ID)
+			filteredTests = append(filteredTests, test)
+		}
+	}
+
+	return filteredTests
+}
+
+func filterAndMergeTests(allTests []CITestConf, workflow, testType, base64Tests string) ([]CITestConf, error) {
+	decodedBytes, err := base64.StdEncoding.DecodeString(base64Tests)
+	if err != nil {
+		return nil, err
+	}
+	var decodedTests []CITestConf
+	err = yaml.Unmarshal(decodedBytes, &decodedTests)
+	if err != nil {
+		return nil, err
+	}
+
+	idFilter := make(map[string]CITestConf)
+	for _, dt := range decodedTests {
+		idFilter[dt.ID] = dt
+	}
+
+	var filteredTests []CITestConf
+	for _, test := range allTests {
+		workflowMatch := workflow == "" || contains(test.Workflows, workflow)
+		typeMatch := testType == "" || test.TestType == testType
+
+		if decodedTest, exists := idFilter[test.ID]; exists && workflowMatch && typeMatch {
+			// Override test inputs from the base64 encoded tests
+			for k, v := range decodedTest.TestInputs {
+				if test.TestInputs == nil {
+					test.TestInputs = make(map[string]string)
+				}
+				test.TestInputs[k] = v
+			}
+			test.IDSanitized = sanitizeTestID(test.ID)
+			filteredTests = append(filteredTests, test)
+		}
+	}
+
+	return filteredTests, nil
+}
+
+func sanitizeTestID(id string) string {
+	// Define a regular expression that matches any character not a letter, digit, hyphen
+	re := regexp.MustCompile(`[^a-zA-Z0-9-_]+`)
+	// Replace all occurrences of disallowed characters with "_"
+	return re.ReplaceAllString(id, "_")
+}
+
+// Utility function to check if a slice contains a string.
+func contains(slice []string, element string) bool {
+	for _, s := range slice {
+		if s == element {
+			return true
+		}
+	}
+	return false
+}
+
+// filterCmd represents the filter command
+var filterCmd = &cobra.Command{
+	Use:   "filter",
+	Short: "Filter test configurations based on specified criteria",
+	Long: `Filters tests from a YAML configuration based on name, workflow, test type, and test IDs.
+Example usage:
+./e2e_tests_tool filter --file .github/e2e-tests.yml --workflow "Run Nightly E2E Tests" --test-type "docker" --test-ids "test1,test2"`,
```

Comment:
@lukaszcl if I have a workflow that runs tests in this manner: `smoke/ocr_test.go:*` and I add a new test to `smoke/ocr_test.go`, will that filtering tool understand that my new test is not dead, because `smoke/ocr_test.go:*` is used? or does it require test name to be present more explicitly?